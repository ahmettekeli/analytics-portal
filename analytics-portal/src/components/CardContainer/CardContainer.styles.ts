import styled from "styled-components";

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-width: 100%;
`;

export const Container = styled.div`
  width: 100%;
  margin: 0.5rem;
`;
