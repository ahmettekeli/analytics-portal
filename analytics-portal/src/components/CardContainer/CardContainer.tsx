import { useEffect, useState } from "react";
import { useAPI } from "context/Store";
import LinearProgress from "@material-ui/core/LinearProgress";
import Snackbar from "@material-ui/core/Snackbar";
import useNotification from "useNotification";
import CardList from "../CardList/CardList";
import * as S from "./CardContainer.styles";

function CardContainer() {
  const {
    state: { appData, isLoading, errorMessage },
  } = useAPI();

  const {
    visible: alertVisibility,
    hideNotification,
    showNotification,
  } = useNotification();
  const [activeApps, setActiveApps] = useState<typeof appData>([]);
  const [inactiveApps, setInactiveApps] = useState<typeof appData>([]);

  useEffect(() => {
    errorMessage && showNotification();
  }, [errorMessage]);

  useEffect(() => {
    if (!isLoading) {
      //dividing active and inactive campaigns into their own array.
      appData.forEach((item) => {
        if (item.active) {
          setActiveApps((prev) => [...prev, item]);
        } else {
          setInactiveApps((prev) => [...prev, item]);
        }
      });
    }
  }, [appData, isLoading]);

  if (isLoading) {
    return <LinearProgress data-testid="loading-bar" />;
  }

  return (
    <S.Wrapper data-testid="card-container">
      <CardList type="active" apps={activeApps}></CardList>
      <S.StyledLine width="80%" data-testid="app-seperation-line" />
      <CardList type="inactive" apps={inactiveApps}></CardList>
      {/* <Snackbar
        open={alertVisibility}
        autoHideDuration={2500}
        onClose={hideNotification}
        message={errorMessage}
      /> */}
    </S.Wrapper>
  );
}

export default CardContainer;
