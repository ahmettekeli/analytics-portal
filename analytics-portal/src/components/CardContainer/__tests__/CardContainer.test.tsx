import { cleanup, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import CardContainer from "../CardContainer";
import AnalyticsProvider from "context/Store";

describe("CardContainer", () => {
  let renderResult: ReturnType<typeof render>;
  beforeEach(() => {
    renderResult = render(
      <AnalyticsProvider>
        <CardContainer />
      </AnalyticsProvider>
    );
  });
  afterEach(cleanup);

  test("Shows loading bar on loading", () => {
    expect(renderResult.getByTestId("loading-bar")).toBeInTheDocument();
  });

  //TODO have context store mock with isLoading false;
  // test("Renders CardContainer component", () => {
  //   expect(renderResult.getByTestId("card-container")).toBeInTheDocument();
  // });

  // test("Renders seperation line between active-inactive apps", () => {
  //   expect(renderResult.getByTestId("app-seperation-line")).toBeInTheDocument();
  // });

  test("CardContainer matches snapshot", () => {
    const tree = renderer
      .create(
        <AnalyticsProvider>
          <CardContainer />
        </AnalyticsProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
