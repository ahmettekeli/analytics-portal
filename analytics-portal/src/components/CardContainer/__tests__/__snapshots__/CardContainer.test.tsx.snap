// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer CardContainer matches snapshot 1`] = `
<div
  className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
  data-testid="loading-bar"
  role="progressbar"
>
  <div
    className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
    style={Object {}}
  />
  <div
    className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
    style={Object {}}
  />
</div>
`;
